### PR TITLE
Blog Scaladoc tables

### PR DESCRIPTION
**Pull request by janekdb**

### CODE REVIEW

**Comment by janekdb** [comment]
Despite the branch name there is no part two. This is it!

File: blog/_posts/2018-10-03-scaladoc-tables.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+<img alt="Example Table 1"
+     src="/resources/img/blog/scaladoc-tables/example-table-1.png">
+
+On an incidental note, there is no requirement to align header and cell markdown pipes. This is done in the examples
+to make them easier to scan.
+
+
+### GitHub Flavored Markdown Conformance
+
+Scaladoc tables support the [Github Flavored Markdown Spec Table Extension](https://github.github.com/gfm/#tables-extension-).
```
[the remark this answers is not in the document]
Comment (janekdb) [reply blog/_posts/2018-10-03-scaladoc-tables.md]: 👍

File: blog/_posts/2018-10-03-scaladoc-tables.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+post-type: blog
+by: Janek Bogucki
+title: "Introducing Scaladoc Tables"
+---
+
+Scala 2.12.7 adds support for markdown tables in Scaladoc comments.
+
+### A Simple Example
+
+The markdown for tables follows Github Markdown Flavored markdown with some restrictions.
```
[the remark this answers is not in the document]
Comment (janekdb) [reply blog/_posts/2018-10-03-scaladoc-tables.md]: 👍

File: blog/_posts/2018-10-03-scaladoc-tables.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+*/
+trait Bibliography
+```
+
+Markdown paid, we get this reward,
+
+<img alt="Example Table 2"
+     src="/resources/img/blog/scaladoc-tables/example-table-2.png">
+
+Breaking content over several lines is not supported in GHFM tables. Use a `<br/>` tag or go with HTML tables if you have needs beyond the simple tables supported by GHFM.
```
[the remark this answers is not in the document]
Comment (janekdb) [reply blog/_posts/2018-10-03-scaladoc-tables.md]: 👍

File: blog/_posts/2018-10-03-scaladoc-tables.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+| Delimiter Cell Content | Alignment Type |
+| ---     | --- |
+| `---`   | Left |
+| `:---`  | Left |
+| `---:`  | Right |
+| `:---:` | Center |
+
+### Including pipe (`|`) in content.
+
+Scala 2.12.8 supports the [escaping of the cell delimiter in cell content](https://github.com/scala/scala/pull/7247) with `\|`.
```
[the remark this answers is not in the document]
Comment (janekdb) [reply blog/_posts/2018-10-03-scaladoc-tables.md]: 👍

**Comment by janekdb** [comment]
@Philippus All your comments have been addressed. Kindly review.

**Comment by SethTisue** [comment]
Perhaps the post should say something, near the end, about what Scala 3 / Dotty supports or will support?

File: blog/_posts/2018-10-03-scaladoc-tables.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+A table is comprised of,
+
+ - A header row
+ - A delimiter row
+ - Zero or more data rows
+
+This is how the above markdown renders,
+
+<img alt="Example Table 1"
+     src="/resources/img/blog/scaladoc-tables/example-table-1.png">
```
Comment (ashawley) [inline blog/_posts/2018-10-03-scaladoc-tables.md]: Jekyll supports an image syntax

```markdown
![Example Table 1](/resources/img/blog/scaladoc-tables/example-table-1.png)
```

Comment (janek-inferess) [reply blog/_posts/2018-10-03-scaladoc-tables.md]: @ashawley Thanks! Changes made to both images.

**Comment by janek-inferess** [comment]
> Perhaps the post should say something, near the end, about what Scala 3 / Dotty supports or will support?

@SethTisue If I can be supplied with the text I can add it now, otherwise I'll need to research it.

**Comment by janekdb** [comment]
> Perhaps the post should say something, near the end, about what Scala 3 / Dotty supports or will support?

@SethTisue If I can be supplied with the text I can add it now, otherwise I'll need to research it.

**Comment by ashawley** [comment]
This is exciting news and very good documentation, but maybe most of the material should be added to https://docs.scala-lang.org/overviews/scaladoc/for-library-authors.html

The images don't render very well on this site, maybe they'll look better on docs.scala-lang.org?

**Comment by ashawley** [comment]
The first one was the one I was more concerned with:

screen shot 2018-10-03 at 12 30 53 pm

**Comment by janekdb** [comment]
@ashawley Do you know how to turn off resizing of images?

**Comment by ashawley** [comment]
I looked, but couldn't figure it out.

**Comment by janekdb** [comment]
I don't intend to look into the `jekyll` image control so reviewers please list any further changes required.

**Comment by ashawley** [comment]
It's a CSS issue with the new site, and not intrinsically a Jekyll issue.

**Comment by SethTisue** [comment]
> If I can be supplied with the text I can add it now, otherwise I'll need to research it.

see 
https://github.com/scala/scala/pull/6043#issuecomment-393106385

File: blog/_posts/2018-10-03-scaladoc-tables.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
+---
+layout: blog-detail
+post-type: blog
+by: Janek Bogucki
+title: "Introducing Scaladoc Tables"
+---
+
+Scala 2.12.7 adds support for markdown tables in Scaladoc comments.
```
Comment (ashawley) [inline blog/_posts/2018-10-03-scaladoc-tables.md]: Will it be added in 2.13?  Library authors who cross-compile on 2.11 should avoid for now?

Comment (ashawley) [reply blog/_posts/2018-10-03-scaladoc-tables.md]: Library authors should only publish 2.12 docs or later?

**Comment by janekdb** [comment]
@SethTisue I've gone dotty over dotty...

image

File: blog/_posts/2018-10-03-scaladoc-tables.md (hunk 6 above)
Comment (janekdb) [reply blog/_posts/2018-10-03-scaladoc-tables.md]: It's been merged to 2.13.x IIRC. I feel advising not to use a feature that is only present from 2.12 onwards when cross compiling to 2.11 is overly prescriptive so I vote to not mention this.

File: blog/_posts/2018-10-03-scaladoc-tables.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+| `---:`  | Right |
+| `:---:` | Center |
+
+### Including pipe (`|`) in content.
+
+Scala 2.12.8 will support the [escaping of the cell delimiter in cell content](https://github.com/scala/scala/pull/7247) with `\|`.
+
+### More Examples
+
+For a smorgasbord of examples take a look at [examples](http://janekdb.github.io/scala/PR-7247/scala/test/scaladoc/tables/code/index.html).
```
Comment (Philippus) [inline blog/_posts/2018-10-03-scaladoc-tables.md]: quite an expensive rookworst :)

Comment (janekdb) [reply blog/_posts/2018-10-03-scaladoc-tables.md]: Rookworsts on me!

image

Not actually me.

File: blog/_posts/2018-10-03-scaladoc-tables.md (hunk 6 above)
Comment (ashawley) [reply blog/_posts/2018-10-03-scaladoc-tables.md]: Agree, doesn't seem like there's a reason to dissuade library authors, but maybe just mention that "library authors who use the table syntax will need to only publish from 2.12"?

Comment (SethTisue) [reply blog/_posts/2018-10-03-scaladoc-tables.md]: > It's been merged to 2.13.x IIRC

(it has, but after M5 was released)

Comment (janekdb) [reply blog/_posts/2018-10-03-scaladoc-tables.md]: @ashawley I mentioned the graceful degradation forward compatibility with 2.11 in a new section.

**Comment by janekdb** [comment]
@SethTisue @ashawley @Philippus - All changes addressed. Last change was to fix up some wonky wording in the Scala 2.11 and dotty sections. All done. Ready for review.

File: blog/_posts/2018-10-04-scaladoc-tables.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+Scala 2.12.8 will support the [escaping of the cell delimiter in cell content](https://github.com/scala/scala/pull/7247) with `\|`.
+
+### More Examples
+
+For a smorgasbord of examples take a look at [examples](http://janekdb.github.io/scala/PR-7247/scala/test/scaladoc/tables/code/index.html).
+
+### Dotty and Scala 3
+
+With `dotty` now boosting it's share of the development horizon it's important to place GFM into a [dottydoc](https://dotty.epfl.ch/docs/usage/dottydoc.html) and hence Scaladoc 3 context.
```
Comment (Philippus) [inline blog/_posts/2018-10-04-scaladoc-tables.md]: its share

File: blog/_posts/2018-10-04-scaladoc-tables.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+In overview `dottydoc` supports the currrent Scaladoc 2.12 MediaWiki flavored markdown, but it also supports more contemporary flavors of markdown.
```
Comment (Philippus) [inline blog/_posts/2018-10-04-scaladoc-tables.md]: typo: currrent -> current

File: blog/_posts/2018-10-04-scaladoc-tables.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+
+Within the realm of tables markdown `dottydoc` supports some extensions via the [flexmark-java table extension](https://github.com/vsch/flexmark-java/wiki/Tables-Extension).
+
+[flexmark-java](https://github.com/vsch/flexmark-java) allow tables to be defined using a more expressive variant of GFM table markdown. Tables you create now will have a route to being reused for `dottydoc` API documentation.
```
Comment (Philippus) [inline blog/_posts/2018-10-04-scaladoc-tables.md]: allows

Comment (janekdb) [reply blog/_posts/2018-10-04-scaladoc-tables.md]: 👍

**Comment by janekdb** [comment]
@Philippus All requested changes made.

File: blog/_posts/2018-10-04-scaladoc-tables.md [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+post-type: blog
+by: Janek Bogucki
+title: "Introducing Scaladoc Tables"
+---
+
+Scala 2.12.7 adds support for markdown tables in Scaladoc comments.
+
+### A Simple Example
+
+The markdown for tables follows [GitHub Markdown Flavored Markdown](https://github.github.com/gfm/) with some restrictions.
```
Comment (Philippus) [inline blog/_posts/2018-10-04-scaladoc-tables.md]: GitHub Flavored Markdown

Comment (janekdb) [reply blog/_posts/2018-10-04-scaladoc-tables.md]: 👍

Comment (janekdb) [reply blog/_posts/2018-10-04-scaladoc-tables.md]: Fix pushed

**Comment by janekdb** [comment]
@Philippus Thanks for close review.

@SethTisue - No pending review comments.

**Comment by janekdb** [comment]
Pushed one more fix
`Within the realm of tables markdown` -> `Within the realm of table markdown`

**Comment by SethTisue** [comment]
https://twitter.com/scala_lang/status/1047853269630902274

🎉

**Comment by janekdb** [comment]
Yay!

**Pull request merged**